cclib/bridge/cclib2horton.py

```diff
@@ -80,7 +80,53 @@ def makehorton(data):
             attributes["npa_charges"] = data.atomcharges["natural"]
 
     elif hortonver == 3:
-        pass  # Populate with bridge for horton 3 in later PR
+        # In horton 3 IOData, inputs are type-checked -- thus the bridge also verifies the types.
+        # if coordinates are known; numpy.ndarray (size natom-by-3)
+        if hasattr(data, "atomcoords"):
+            attributes["atcoords"] = numpy.asanyarray(data.atomcoords)[-1]
+        # if atomic numbers are known; numpy.ndarray (size natom)
+        if hasattr(data, "atomnos"):
+            attributes["atnums"] = numpy.asanyarray(data.atomnos)
+        # if orbital coefficients known; iodata.orbitals.MolecularOrbitals
+        if hasattr(data, "mocoeffs"):
+            moattr = {
+                "kind": "restricted",
+                "norba": len(data.mocoeffs[0]),
+                "norbb": None,
+                "occs": numpy.concatenate(
+                    numpy.ones(data.homos[0]), numpy.zeros(len(data.mocoeffs[0]) - data.homos[0])
+                ),
+                "coeffs": data.mocoeffs[0],
+                "energies": None,
+                "irreps": None,
+            }
+            # and if unrestricted:
+            if len(mocoeffs) == 2:
+                moattr.kind = "unrestricted"
+                moattr.norbb = len(data.mocoeffs[1])
+                moattr.coeffs.append(data.mocoeffs[1].T)
+                moattr.occs.append(
+                    numpy.concatenate(
+                        numpy.ones(data.homos[1]),
+                        numpy.zeros(len(data.mocoeffs[1]) - data.homos[1]),
+                    )
+                )
+            attributes["mo"] = MolecularOrbitals(**moattr)
+        # if multiplicity known; float / should not be set when mocoeffs present
+        # Refer to IOData code:
+        # https://github.com/theochem/iodata/blob/b36513d162f99b57264005583701c6987037839c/iodata/iodata.py#L174
+        if (
+            hasattr(data, "mult")
+            and data.mult != None
+            and not isinstance(data.mocoeffs, numpy.ndarray)
+        ):
+            attributes["spinpol"] = data.mult - 1  # horton has 2S+1, iodata has 2S
+        # if pseudopotentials exist; numpy.ndarray (size natom)
+        if hasattr(data, "coreelectrons") and isinstance(data.coreelectrons, numpy.ndarray):
+            attributes["atcorenums"] = data.atomnos - data.coreelectrons
+        # if mulliken charges are known; dict of numpy.ndarrays (size natom)
+        if hasattr(data, "atomcharges") and isinstance(data.atomcharges, dict):
```

Comment:
Another redundant check.

Reply:
I've removed the unnecessary checks. Thanks.